### ISSUE TITLE
 bug(cdk/menu): Trigger does not update ARIA values correctly in OnPush components (Accessibility)

### ISSUE THREAD

**Issue by kryoussef** [angular/components#29041]
### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### Description

It seems like the issue : [27725](https://github.com/angular/components/issues/27725) has not been resolved,
The Trigger does not update aria values and misbehaves in OnPush components.

### Reproduction

StackBlitz link: https://stackblitz.com/edit/angular-bqrgp1-fjdmrr?file=src%2Fapp%2Fmenu-overview-example.ts
Steps to reproduce:
1. use NVDA Speech viewer
2. open the menu and select items (multiple times)
3. ![screen reader](https://github.com/angular/components/assets/168361048/dcb419eb-be1e-4b4b-9c81-f3430d33931c)

### Expected Behavior

When selecting an item `menu-item` from the menu, the `aria-label` and `aria-expanded` should be updated  correctly, 
From the example above, NVDA should read item2 with menu being collapsed

### Actual Behavior

As shown in the screenshot, when we click on the item2, NVDA is reading item1 with the menu being expanded

### Environment

- Angular: 17.3.7
- CDK/Material: 17.3.7
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

**Comment by crisbeto** [comment]
I'm a bit confused about the setup here. It's using `MatMenu` even though the issue is talking about the CDK menu. Also the bindings you're mentioning are set up by your own component, not to Material/CDK ones so you'll have to call `markForCheck` there yourself.

**Comment by kryoussef** [comment]
HEllo @crisbeto , 
The issue is : [26262](https://github.com/angular/components/issues/26262) and the fixe is [26279](https://github.com/angular/components/pull/26279)
I will have to update the issue description

**Comment by crisbeto** [comment]
I spent some time trying to reproduce it, but the `aria-expanded` was being updated as expected in the Stackblitz. Could there be something in your app that's interfering with it?

**Comment by kryoussef** [comment]
Hello, as shown in this video, the aria-label is not  updated and for accessibility, it is a problem,
here is what NVDA reads while clocking or pressing enter on the menu trigger.
https://github.com/angular/components/assets/168361048/cef03f46-a65f-4804-baa1-3d222bd88a26

**Comment by kryoussef** [comment]
Hello @crisbeto , any update on this issue ?

**Comment by kryoussef** [comment]
Hello @crisbeto, 

Can I have an update for this issue ? I am waiting for your feedback, please find above the video reproducing the bug

**Comment by mmalerba** [comment]
@adolgachev assigning to you since you've been working on a11y, are you able to reproduce this?
